consensus/state.go

```diff
@@ -1325,7 +1333,8 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 	// NOTE: the proposal signature is validated when it is received,
 	// and the proposal block parts are validated as they are received (against the merkle hash in the proposal)
 	logger.Debug("prevote step: ProposalBlock is valid")
-	cs.signAddVote(cmtproto.PrevoteType, cs.ProposalBlock.Hash(), cs.ProposalBlockParts.Header())
+	cs.addPrevotedHash(proposalBlockHash)
```

Comment:
May be unnecessary since the below `cs.signAddVote(...)` will push to `cs.internalMsgQueue` which will be handled by `cs.handleMsg(...)` which will see the prevote, realize it comes from the current validator, and call `cs.ddPrevotedHash(...)`